src/plots/cartesian/axes.js

```diff
@@ -1781,11 +1779,12 @@ axes.drawOne = function(gd, ax, opts) {
     if(ax.type === 'multicategory') {
         var labelLength = 0;
         var pad = {x: 2, y: 10}[axLetter];
```

Comment:
I didn't understand why different padding were needed for x and y directions? It is not related to this PR though.

Reply:
True. Most of the texts are horizontal so that's OK to have a stronger divider. But what if the width of image was less than the height?

Reply:
I agree, because otherwise the labels run into each other in the reading direction. Which says you might want the larger padding on x too, if both levels of label are rotated.

Reply:
this line:

https://github.com/plotly/plotly.js/blob/e1ac60b1078b728277f2ba887cf5ad98b050f77b/src/plots/cartesian/axes.js#L1787

appears to do an ok job:

```js
var l = 'loooooooooong'
var l2 = 'looooooooooo oooooong'

Plotly.newPlot(gd, [{
  type: 'bar',
  x: [
    [l, l, l2, l2],
    [l, l2, l, l2]
  ],
  y: [1, 2, 3, 1]
}], {
  xaxis: {automargin: true},
  width: 300,
  height: 700
})
```

![newplot 15](https://user-images.githubusercontent.com/6675409/50016398-068f4f80-ff97-11e8-84d5-7b842cf99176.png)

Reply:
Anyway, I'll revisit this problem when fixing https://github.com/plotly/plotly.js/issues/2434 - which I hope I'll get to soon :crossed_fingers: